pymeasure/instruments/toptica/adapters.py

```diff
@@ -0,0 +1,132 @@
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import re
+import time
+
+from pymeasure.adapters import VISAAdapter
+
+
+class TopticaAdapter(VISAAdapter):
+    """ Adapter class for connecting to Toptica Console via a serial
+    connection.
+
+    :param port: pyvisa resource name of the instrument
+    :param baud_rate: communication speed, defaults to 115200
+    :param kwargs: Any valid key-word argument for VISAAdapter
+    """
+    # compiled regular expression for finding numerical values in reply strings
+    _reg_value = re.compile(r"\w+\s+=\s+(\w+)")
+
+    def __init__(self, port, baud_rate, **kwargs):
+        kwargs.setdefault('preprocess_reply', self.extract_value)
+        super().__init__(port, **kwargs)
+        # hack to set baud_rate since VISAAdapter filters it out when spcified
+        # as keyword argument. issue #334
+        self.connection.baud_rate = baud_rate
+        # configure communication mode
+        super().write('echo off')
+        super().write('prom off')
+        time.sleep(0.04)
+        # clear the initial messages from the controller
+        self.flush()
+        self.ask('talk usual')
+
+    def extract_value(self, reply):
+        """ preprocess_reply function which tries to extract <value> from 'name
+        = <value> [unit]'. If <value> can not be identified the orignal string
+        is returned.
+
+        :param reply: reply string
+        :returns: string with only the numerical value, or the original string
+        """
+        r = self._reg_value.search(reply)
+        if r:
+            return r.groups()[0]
+        else:
+            return reply
+
+    def check_acknowledgement(self, reply):
+        """ checks if reply is '[OK]', otherwise a ValueError is raised and the
+        read buffer is flushed because one has to assume that some
+        communication is out of sync.
+
+        :param reply: reply string of the instrument which should be checked
+        """
+        if reply != '[OK]':
+            self.flush()
+            raise ValueError(
+                f"TopticaAdapter: Error after command '{self.lastcommand}' "
+                f"with message '{reply}'")
+
+    def _read_stripped(self):
```

Comment:
are you sure this is necessary? According to the [docs](https://pyvisa.readthedocs.io/en/1.8/resources.html?highlight=read_termination#termination-characters) should already strip `read_termination` for you.

Reply:
I forgot to set read_termination and by default its empty. will fix this and remove this function